### PR TITLE
API improvements and better error handling

### LINKED ISSUE #38: Error handling/formatting
For some reason the errors we return to clients cause them to barf parsing JSON. I think @bobcallaway fixed this in rekor with typed error messages, but I can't remember how.

### PR DESCRIPTION
**Pull request by bobcallaway**
This adds in more explicit error handling in the API, as well as:

- changing response type of /signingCert to
  application/pem-certificate-chain
- uses oidc.IDToken as authentication principal in go-swagger

Fixes #38

Signed-off-by: Bob Callaway <bcallawa@redhat.com>

### CODE REVIEW

File: pkg/api/signing_cert.go [hunk 1]
```diff
@@ -17,66 +17,74 @@ limitations under the License.
...
 	"github.com/sigstore/fulcio/pkg/log"
+	"github.com/sigstore/fulcio/pkg/oauthflow"
 	"github.com/spf13/viper"
-	"golang.org/x/net/context"
 )
 
-func SigningCertHandler(params operations.SigningCertParams, principal interface{}) middleware.Responder {
+func SigningCertHandler(params operations.SigningCertParams, principal *oidc.IDToken) middleware.Responder {
+	ctx := params.HTTPRequest.Context()
+	emailAddress, _, _ := oauthflow.EmailFromIDToken(principal)
```
Comment (dlorenc) [inline pkg/api/signing_cert.go:40]: do we need to handle errors here?

**Review by dlorenc — APPROVED** [review]
Just the one nit about errors

File: pkg/api/signing_cert.go (hunk 1 above)
Comment (bobcallaway) [reply pkg/api/signing_cert.go:40]: In theory, no. This code path should only be called after the request was authenticated, which includes validation of the ID token.

Comment (bobcallaway) [reply pkg/api/signing_cert.go]: added code in next commit to be defensive here

**Review by dlorenc — APPROVED** [review]

**Pull request merged**